src/collections/Menu/Menu.js

```diff
@@ -81,10 +81,11 @@ export default class Menu extends Component {
 
     return Children.map(children, (child, i) => {
       const isItem = child.type === MenuItem
+      const isLink = _.has(child, 'props.href') || _.has(child, 'props.link') || _.has(child, 'props.onClick')
 
       if (isItem) {
         const onClick = (e) => {
-          this.handleItemClick(e, i)
+          if (isLink) this.handleItemClick(e, i)
           if (child.props.onClick) child.props.onClick(e, i)
         }
```

Comment:
I'm proposing two separate usage patterns.

## items={} prop

My suggestion is that this usage will automatically manage the active item state:

```jsx
<Menu items={[{}, {}, ...]} />
```

With this usage we can add the onClick handlers our selves and manage the `active` prop on the `MenuItem`s internally.  We can do this easily since we must map over all the `items` to create the `MenuItems`.

## Subcomponents

This usage would not self manage the active item.  The user needs to implement their own `onClick` handlers and set the `active`:

```jsx
class MenuExample extends Component {
  handleItemClick = (index) => this.setState({ activeIndex: index })

  render() {
    const { activeIndex } = this.state
    return (
      <Menu>
        <Menu.Item active={activeIndex === 0} onClick={this.handleItemClick.bind(0)} />
        <Menu.Item active={activeIndex === 1} onClick={this.handleItemClick.bind(1)} />
        <Menu.Item active={activeIndex === 2} onClick={this.handleItemClick.bind(2)} />
      </Menu>
    )
  }
}
```

They can obviously use any convention they want for this.  It doesn't have to be an activeIndex:

```jsx
class MenuExample extends Component {
  handleItemClick = (name) => this.setState({ activeItem: name })

  render() {
    const { activeItem } = this.state
    return (
      <Menu>
        <Menu.Item active={activeItem === 'home'} onClick={this.handleItemClick.bind('home')} />
        <Menu.Item active={activeItem === 'search'} onClick={this.handleItemClick.bind('search')} />
        <Menu.Item active={activeItem === 'settings'} onClick={this.handleItemClick.bind('settings')} />
      </Menu>
    )
  }
}
```

***

These ideas can be improved on, but I think we should only auto manage the active state if you use the `items` prop.